### ISSUE TITLE
Non-compliant html props make theme incompatible with React.js

### ISSUE THREAD

**Issue by WorldsEndless** [byuweb/byu-theme-components#334]
# Description

We build apps with react.js and would like to use the BYU theme. However, the shadow dom functionality hinges on illegal html props and properly compliant libraries like React.js disallow the rendering of these html elements. I run in to this problem, for example on the property:

    <span slot="site-title">
    
"Slot" is illegal and React won't render it. Legal would be 

    <span data-slot="site-title">

# Issue Type

*Is this (add an x in the boxes that apply)*

- [ ] A difference between the components and the [Official Specification](https://brand.byu.edu)?
- [x] A bug, such as a Javascript error, or the UI not rendering properly
    on a page?
- [ ] Inconsistent appearance/behavior between browsers?
- [x] An issue on mobile browsers?
- [x] A request for a new feature/enhancement?

# Browsers Affected

*Add an x in all the boxes that apply. Please mark desktop and mobile
browsers separately.*

**We support the last two versions of Chrome, Firefox, Safari, and Edge,
plus Internet Explorer 11.**

## Desktop Browsers

- [x] Google Chrome
- [x] Mozilla Firefox
- [x] Apple Safari
- [x] Microsoft Edge
- [x] Microsoft Internet Explorer 11
- [x] Other (please specify)

## Mobile Browsers

- [x] Safari for iOS
- [x] Chrome for iOS
- [x] Firefox Mobile for Safari
- [x] Chrome for Android
- [x] Firefox Mobile for Android
- [x] Other (please specify)

# Web Site Platform

*What is hosting your website?*

- [ ] Drupal 7
- [ ] Drupal 8
- [ ] Wordpress
- [x] Custom Site
- [ ] Don't Know

**Comment by ThatJoeMoore** [comment]
Actually, `slot` is part of the DOM specification.  See https://html.spec.whatwg.org/multipage/scripting.html#the-slot-element.  React has supported this part of the DOM spec since https://github.com/facebook/react/pull/8061 .

**Comment by ThatJoeMoore** [comment]
Looks like, even though that PR was merged a year ago, it didn't actually ship until React 16 in September.

Why anyone would want to use a framework that actively restricts what you can use on an element is beyond me.

**Comment by WorldsEndless** [comment]
Thanks @ThatJoeMoore ; definitely set me on the right track for resolution.

**Comment by ThatJoeMoore** [comment]
Yeah, lemme know if you run into any issues.  It kinda stinks that React does attribute whitelisting and that it took them so long to support `slot` - it first shipped in browsers in 2015. If you're gonna have a whitelist, it better take less than two years to get something added.

If upgrading isn't an option, I found this issue on Skatejs: https://github.com/skatejs/skatejs/issues/1096#issuecomment-299016892

There's a lot of chatter in the comments about switching to Preact, but the comment I linked to has an approach that looks like it might work. It's not as nice as just slapping the attribute on there, but at least it'll work.
